pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/verticalpodautoscaler.go

```diff
@@ -29,6 +33,10 @@ func ExtractVerticalPodAutoscaler(v *v1.VerticalPodAutoscaler) *model.VerticalPo
 // extractVerticalPodAutoscalerSpec converts the Kubernetes spec to our custom one
 // https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1/types.go#L73
 func extractVerticalPodAutoscalerSpec(s *v1.VerticalPodAutoscalerSpec) *model.VerticalPodAutoscalerSpec {
+	if s == nil {
```

Comment:
AFAICT this one should always be there https://github.com/kubernetes/autoscaler/blob/ab2e467452d7786a3cdb655bd26f220eaa8270a0/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1/types.go#L58 (not optional)

Reply:
Do you think this should be removed to match the structure? Is there any harm in additional nil checks?

Reply:
[106fdd9](https://github.com/DataDog/datadog-agent/pull/15488/commits/106fdd9fd3c5cdb59ed41817e2ff346bd3319168)

Reply:
> Is there any harm in additional nil checks?

IMO, without nil check, we can find issue earlier if we have nil spec while spec shouldn't be nil